# `jsdoc-full-sentences`

Enforces complete sentences for JSDoc comments.

## Rule Details

```ts
// Bad
/** Sometimes I forget to put a period on my comments */

// Good
/** Sometimes I forget to put a period on my comments. */
```

## Options

```json
{
  "rules": {
    "isaacscript/jsdoc-full-sentences": "error"
  }
}
```

This rule is not configurable.
