# `jsdoc-description-newline`

Requires a newline between the JSDoc description and tags, if any.

## Rule Details

```ts
// Bad

// Good
```

## Options and Defaults

```json
{
  "rules": {
    "isaacscript/jsdoc-description-newline": "error"
  }
}
```

This rule is not configurable.

## Resources

- [Rule source](../../src/rules/jsdoc-description-newline.ts)
- [Test source](../../tests/rules/jsdoc-description-newline.test.ts)
