# `jsdoc-code-block`

Requires a language for every JSDoc code block.

## Rule Details

```ts
// Bad

// Good
```

## Options and Defaults

```json
{
  "rules": {
    "isaacscript/jsdoc-code-block": "error"
  }
}
```

This rule is not configurable.

## Resources

- [Rule source](../../src/rules/jsdoc-code-block.ts)
- [Test source](../../tests/rules/jsdoc-code-block.test.ts)
